pkg/machines/reducers.es6

```diff
@@ -189,13 +189,12 @@ function systemInfo(state, action) {
     switch (action.type) {
     case UPDATE_OS_INFO_LIST: {
         if (action.osInfoList instanceof Array) {
-            state.osInfoList = [...action.osInfoList];
+            return Object.assign({}, state, { osInfoList: action.osInfoList });
```

Comment:
So the idea is that `systemInfo()` must never mutate `state`, but return a new object? I'm afraid I don't know much about how `lazyComposedReducer()` works, but at first sight it looks like `parentReducer()` would already create a new object which you can then mutate?

Reply:
We should not mutate state inside systemInfo or any other state handling function. We should return always a new object.
```lazyComposedReducer``` and ```parentReducer``` and all these functions are not involved in the systemInfo  it's affecting only config and I am not touching the config function in this PR. 
https://github.com/cockpit-project/cockpit/blob/8aeeea2d6b6e775b70921977c05e40fa1d2bdeb4/pkg/machines/reducers.es6#L333
Systeminfo function is not mixing with lazyComposedreducer IOW.